portfolio/src/main/java/com/google/sps/servlets/LoginInfo.java

```diff
@@ -0,0 +1,66 @@
+package com.google.sps.servlets;
+
+import java.util.Objects;
+
+/**
+ * Store data for user login.
+ */
+public class LoginInfo {
+  private boolean loggedIn;
+  private String userName;
+
+  public LoginInfo() {
+    loggedIn = false;
+    userName = "";
+  }
+
+  public LoginInfo(boolean loggedIn, String userName) {
+    this.loggedIn = loggedIn;
+    this.userName = userName;
+  }
+
+  public boolean isLoggedIn() { return this.loggedIn; }
+
+  public boolean getLoggedIn() { return this.loggedIn; }
+
+  public void setLoggedIn(boolean loggedIn) { this.loggedIn = loggedIn; }
+
+  public String getUserName() { return this.userName; }
+
+  public void setUserName(String userName) { this.userName = userName; }
+
+  public LoginInfo loggedIn(boolean loggedIn) {
+    this.loggedIn = loggedIn;
+    return this;
+  }
+
+  public LoginInfo userName(String userName) {
+    this.userName = userName;
+    return this;
+  }
```

Comment:
This is a lot of boilerplate for a simple class. Do you really need all of these methods? It looks a little like you're trying to anticipate every way someone might want to interact with this class, but an API is often easier to understand and use if there's one clear way use it.

Reply:
Usually i autogenerate these basic methods when creating a class, I agree in this case it seems excessive.